src/client/httpClient.ts

```diff
@@ -114,12 +128,16 @@ export class HTTPClient {
     } else {
       throw new Error('baseUrl required for initialization');
     }
-
-    if (!options.applicationToken) {
-      //TODOV3 Replace this with auth call to get logged in user by providing username and password
-      throw new Error('applicationToken required for initialization');
-    } else{
+    this.username = options.username || '';
+    this.password = options.password || '';
+    this.authUrl = options.authUrl || this.baseUrl + AUTH_API_PATH;
+    if (options.applicationToken) {
       this.applicationToken = options.applicationToken;
+    }else{      
+      if(this.username === '' || this.password === ''){
```

Comment:
Should also throw this error if `!this.authUrl`. If authUrl is not set, you'll end up getting a `Invalid value 'undefined' for header "X-Cassandra-Token"` error whenever you run your first operation.

Reply:
I have a check before this condition as below to make sure the `authUrl` is always set. I believe that should take care?

`this.authUrl = options.authUrl || this.baseUrl + AUTH_API_PATH;`

Reply:
I set `this.baseUrl + AUTH_API_PATH` as the default because, while connecting to the AstraDB we don't have to change the `baseUrl`, just appending `AUTH_API_PATH` should take care of it.
But, on local or running against open source json api with separate auth endpoint, we will have to provide the `authUrl` separately as a connection option.

Reply:
Also, improved the error handling when no token is present. Please see commit [44f4024](https://github.com/stargate/stargate-mongoose/pull/25/commits/44f40247ed30d9cc6ff94455c94bf9d54c902143).

Please approve if this looks good.